### PR TITLE
Add Thorium states to auto clean out old keys

### PR DESCRIPTION
**Pull request by thatch45**
### What does this PR do?

Adds key and status thorium states and a status beacon. Together these can be used to track presence and then remove old keys
### What issues does this PR fix or reference?
#13141
#33882

### CODE REVIEW

File: doc/topics/thorium/index.rst [hunk 1]
```diff
@@ -46,31 +46,37 @@ looks exactly like the Salt Configuration Management state tree, it is just
...
       '*':
-        - logins
+        - key_clean
 
-The Reg/Check/Act Pattern
-=========================
+Thorium Links to Beacons
+========================
+
+The above example as added in the Carbon release of Salt and makes use of the
```
Comment (s0undt3ch) [inline doc/topics/thorium/index.rst]: was

File: salt/beacons/status.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+'''
+The status beacon is intended to send a basic health check event up to the
+master, this allows for event driven routines based on presence to be set up.
+
+The intention of this beacon is to add the config options to add monitoring
+stats to the health beacon making it a one stp shop for gathering systems
```
Comment (s0undt3ch) [inline salt/beacons/status.py]: One stop shop

File: salt/beacons/status.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    if not isinstance(config, dict):
+        return False, ('Configuration for status beacon must be a dictionary.')
+    return True, 'Valid beacon configuration'
+
+
+def beacon(config):
+    '''
+    Just say that we are ok!
+    '''
+    return [{'tag': '{0}'.format(int(time.time()))}]
```
Comment (s0undt3ch) [inline salt/beacons/status.py]: Shouldn't we use UTC?

File: salt/beacons/status.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+'''
+The status beacon is intended to send a basic health check event up to the
+master, this allows for event driven routines based on presence to be set up.
+
+The intention of this beacon is to add the config options to add monitoring
+stats to the health beacon making it a one stp shop for gathering systems
+health and status data
+'''
```
Comment (rallytime) [inline salt/beacons/status.py]: `.. versionadded:: Carbon`

File: salt/thorium/key.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
+'''
+The key Thorium State is used to apply changes to the accepted/rejected/pending keys
+'''
```
Comment (rallytime) [inline salt/thorium/key.py]: `.. versionadded:: Carbon`

File: salt/thorium/status.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
+'''
+This thorium state is used to track the status beacon events and keep track of
+the active status of minions
+'''
```
Comment (rallytime) [inline salt/thorium/status.py]: `.. versionadded:: Carbon`

File: salt/beacons/status.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
```
Comment (rallytime) [inline salt/beacons/status.py]: This needs a UTF-8 line encoding.

File: salt/beacons/status.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+The status beacon is intended to send a basic health check event up to the
+master, this allows for event driven routines based on presence to be set up.
+
+The intention of this beacon is to add the config options to add monitoring
+stats to the health beacon making it a one stp shop for gathering systems
+health and status data
+'''
+
+# Import python libs
+import time
```
Comment (rallytime) [inline salt/beacons/status.py]: `from __future__ import absolute_path`

File: salt/thorium/key.py [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
```
Comment (rallytime) [inline salt/thorium/key.py]: Line encoding here, too.

File: salt/thorium/key.py [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+The key Thorium State is used to apply changes to the accepted/rejected/pending keys
+'''
+# Import python libs
+import time
```
Comment (rallytime) [inline salt/thorium/key.py]: Needs a `__future__` import.

File: salt/thorium/status.py [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
```
Comment (rallytime) [inline salt/thorium/status.py]: Line encoding here, too :)

File: salt/thorium/status.py [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+This thorium state is used to track the status beacon events and keep track of
+the active status of minions
+'''
+# Import python libs
+import time
```
Comment (rallytime) [inline salt/thorium/status.py]: And a `__future__` import here as well.

**Comment by rallytime** [comment]
@thatch45 There's some related pylint errors. Most of the related errors are Python 3 or file encoding things that I mentioned above, but these two are relevant to this change:
- https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/2971/violations/file/salt/thorium/key.py/
- https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/2971/violations/file/salt/thorium/reg.py/

File: salt/beacons/status.py (hunk 3 above)
Comment (thatch45) [inline salt/beacons/status.py]: yes, I had a brain fart and could not remember how to get the seconds in utc

**Comment by thatch45** [comment]
ok, I think I got all of them

**Comment by thatch45** [comment]
Fixed the lint, man this is embarrassing to come back after not committing for so long!
:)

**Comment by thatch45** [comment]
ok, should be good now

**Pull request merged**